libcontainer/cgroups/fs2/cpu.go

```diff
@@ -13,7 +13,15 @@ import (
 	"github.com/opencontainers/runc/libcontainer/configs"
 )
 
+func isCpuSet(cgroup *configs.Cgroup) bool {
+	return cgroup.Resources.CpuWeight != 0 || cgroup.Resources.CpuMax != ""
+}
+
 func setCpu(dirPath string, cgroup *configs.Cgroup) error {
+	if !isCpuSet(cgroup) {
```

Comment:
Not sure whether this validation is useful.

Reply:
The reason I added it is peculiar -- if someone will be modifying `setCpu()`, adding more fields to set, they will be reminded to also modify `isCpuSet()`. Having this check here _might_ also help to uncover bugs.

Reply:
E.g. someone adds `cpuFoobar` setting to setCpu(), but forgets to modify `isCpuSet()`. They add a unit test to check if `cpuFoobar` is applied, and the test fails! Hope it makes sense.

Reply:
Can be just code comment lines?

Reply:
> Can be just code comment lines?

What do you mean? Add a comment like "if you're modifying this, don't forget to modify isSetCpu"? In my experience, this never works. The added runtime check might actually help to uncover such a bug.

Or do you mean adding a comment stating the reason why this check is added?

Reply:
> Or do you mean adding a comment stating the reason why this check is added?

I have amended the commit telling why is*Set is added to set*:


4. Amend all setFoo() functions to call isFooSet(). While this might
   seem unnecessary, it might actually help to uncover a bug.
   Imagine someone:
    - adds a cgroup.Resources.CpuFoo setting;
    - modifies setCpu() to apply the new setting;
    - but forgets to amend isCpuSet() accordingly <-- BUG

   In this case, a test case modifying CpuFoo will help
   to uncover the BUG. This is the reason why it's added.

Reply:
I mean the former, but not a blocker